File: 자료구조/C++/백준-스택-3986.cpp
# include <iostream>
# include <stack>

using namespace std;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);

	int words;
	cin >> words;

	string input;

	int success, result = 0;

	for (int i = 0; i < words; i++)
	{
		cin >> input;
		stack<char> str;
		success = 1;
		str.push(input[0]);
		for (int idx = 1; idx < input.length(); idx++)
		{
			if (!str.empty() && str.top() == input[idx])
			{
				str.pop();
			}
			else
			{
				str.push(input[idx]);
			}
		}
		if (str.empty()) { result++; }
	}

	cout << result << endl;
	return 0;
}

File: 자료구조/Python/백준-스택-3986.py
def main():
    repeat = int(input())
    total = 0

    for i in range(repeat):
        str = input()
        stack = []
        for idx in range(0, len(str)):
            if len(stack) > 0:
                top = stack.pop()
                if str[idx] == top:
                    continue
                else:
                    stack.append(top)
                    stack.append(str[idx])
            else:
                stack.append(str[idx])
        if len(stack) == 0:
            total += 1

    print(total)

if __name__ == "__main__":
    main()